### PR TITLE
Added priority and due date

### PR DESCRIPTION
**Pull request by divija-palleti**
1. Style Changes.
2. Priority added.
3. Deadline/due date added.
4. Deletion of a non-empty ToDo list added. 

image

5. Display of an error in the case of an empty title for todo_item added
image

### CODE REVIEW

File: app/models/todo_item.rb [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
   belongs_to :todo_list
+
+
+
+
+
+
+
+
+
```
Comment (achheSharma) [inline app/models/todo_item.rb:11]: What for?

File: app/models/todo_item.rb [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+  
+  enum priority: [:high, :medium, :low]
   def completed?
     !completed_at.blank?
```
Comment (achheSharma) [inline app/models/todo_item.rb:14]: use completed_at.present? instead of this negation

File: db/migrate/20200518121043_create_todo_items.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 class CreateTodoItems < ActiveRecord::Migration[6.0]
   def change
     create_table :todo_items do |t|
-      t.string :content
+      t.string :content, default: "Title-1"
```
Comment (achheSharma) [inline db/migrate/20200518121043_create_todo_items.rb:4]: Is it a good idea to give a default value to content?

File: app/models/todo_item.rb (hunk 1 above)
Comment (divija-palleti) [reply app/models/todo_item.rb:11]: It was unintentional, will remove that. Thank you.

File: app/models/todo_item.rb (hunk 2 above)
Comment (divija-palleti) [reply app/models/todo_item.rb:14]: Thank you will look into this.

**Comment by achheSharma** [comment]
Good work :)
Few tips that follow:
Code should have similar indentation throughout the codebase.
There are a lot of places where you've left empty space for no reason and also indentation isn't same across files.

File: db/migrate/20200518121043_create_todo_items.rb (hunk 3 above)
Comment (divija-palleti) [reply db/migrate/20200518121043_create_todo_items.rb:4]: I was trying to display Validation Errors in Views but it shows  - undefined method `errors' for nil:NilClass.

snapshot of the _form.html.erb
Screenshot 2020-05-19 at 1 38 02 PM

todo_items_controller

image


Can you help me with this?

**Pull request merged**